scipy/stats/_stats_py.py

```diff
@@ -6134,27 +6223,86 @@ def ttest_1samp(a, popmean, axis=0, nan_policy='propagate',
     uniform distribution, which *does* have a population mean of 0.5, we would
     mistakenly reject the null hypothesis for one of them.
 
-    """
-    a, axis = _chk_asarray(a, axis)
+    `ttest_1samp` can also compute a confidence interval around the population
+    mean.
 
-    contains_nan, nan_policy = _contains_nan(a, nan_policy)
+    >>> rvs = stats.norm.rvs(size=50, random_state=rng)
+    >>> res = stats.ttest_1samp(rvs, popmean=0)
+    >>> ci = res.confidence_interval(confidence_level=0.95)
+    >>> ci
+    ConfidenceInterval(low=-0.3193887540880017, high=0.2898583388980972)
+
+    The bounds of the 95% confidence interval are the
+    minimum and maximum values of the parameter `popmean` for which the
+    p-value of the test would be 0.05.
+
+    >>> res = stats.ttest_1samp(rvs, popmean=ci.low)
+    >>> np.testing.assert_allclose(res.pvalue, 0.05)
+    >>> res = stats.ttest_1samp(rvs, popmean=ci.high)
+    >>> np.testing.assert_allclose(res.pvalue, 0.05)
+
+    Under certain assumptions about the population from which a sample
+    is drawn, the confidence interval with confidence level 95% is expected
+    to contain the true population mean in 95% of sample replications.
+
+    >>> rvs = stats.norm.rvs(size=(50, 1000), loc=1, random_state=rng)
+    >>> res = stats.ttest_1samp(rvs, popmean=0)
+    >>> ci = res.confidence_interval()
+    >>> contains_pop_mean = (ci.low < 1) & (ci.high > 1)
+    >>> contains_pop_mean.sum()
+    953
 
-    if contains_nan and nan_policy == 'omit':
-        a = ma.masked_invalid(a)
-        return mstats_basic.ttest_1samp(a, popmean, axis, alternative)
+    """
+    a, axis = _chk_asarray(a, axis)
 
     n = a.shape[axis]
     df = n - 1
 
-    d = np.mean(a, axis) - popmean
+    mean = np.mean(a, axis)
+    try:
+        popmean = np.squeeze(popmean, axis=axis)
+    except ValueError as e:
+        raise ValueError("`popmean.shape[axis]` must equal 1.") from e
+    d = mean - popmean
     v = _var(a, axis, ddof=1)
     denom = np.sqrt(v / n)
 
     with np.errstate(divide='ignore', invalid='ignore'):
         t = np.divide(d, denom)
     t, prob = _ttest_finish(df, t, alternative)
 
-    return Ttest_1sampResult(t, prob)
+    # when nan_policy='omit', `df` can be different for different axis-slices
+    df = np.broadcast_to(df, t.shape)
+    # _axis_nan_policy decorator doesn't play well with strings
+    alternative_num = {"less": -1, "two-sided": 0, "greater": 1}[alternative]
+    return Ttest_1sampResult(t, prob, df=df, alternative=alternative_num,
+                             standard_error=denom, estimate=mean)
```

Comment:
```suggestion
    return Ttest_1sampResult(t, prob, df=df[()],
                             alternative=alternative_num,
                             standard_error=denom, estimate=mean)
```

DF is sometimes a scalar array:

```python
from scipy import stats

stats.ttest_1samp([1,2,3],0)
# Ttest_1sampResult(statistic=3.464101615137755, pvalue=0.07417990022744853, df=array(2))
```

Reply:
Good find. Because it's due to the `broadcast_to`, let's do [this](https://github.com/scipy/scipy/pull/16835/files#r953123706).

Reply:
> let's do [this](https://github.com/scipy/scipy/pull/16835/files#r953123706).

Looks good, thanks!